tools/pubsys/src/aws/ami/mod.rs

```diff
@@ -372,20 +418,63 @@ async fn _run(args: &Args, ami_args: &AmiArgs) -> Result<HashMap<String, Image>>
     Ok(amis)
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize, Eq, PartialEq, Hash)]
+pub(crate) struct LaunchPermissionDef {
```

Comment:
This seems like it should be an enum, something like:

```rust
#[derive(Serialize, Deserialize, etc)]
#[serde(rename_all = "lowercase")]
enum LaunchPermissionDef {
    Group(String),
    UserId(String),
    OrganizationArn(String),
    OrganizationalUnitArn(String),
}
```

I'm pretty sure this still [serializes to JSON the way you want](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8ac38bfb19f7e379aec294d3746e54b6).